src/components/timeslot/index.tsx

```diff
@@ -105,6 +105,7 @@ const DEFAULT_TIMESLOT_RECURRENCE: TimeRecurrence = {
 };
 
 export type TimeslotRecurrenceComponentPropsType = {
+  pk?: TimeslotPK;
```

Comment:
Why do you add pk here? It's supposed to be an alias for id.

Reply:
In the backend: "pk" === "id", and is created as the recurrence is saved the first time. We only need one of them in the API for that reason, and we only need one of them in the frontend for that reason, and whatever it is called, it does not exist until first save.

Reply:
The two values are not equal in this case. "pk" is actually the primary key of the Timeslot which the recurrence belongs to. "id" is a value which is assigned to each recurrence in the frontend when rendered. It is used to reference a specific recurrence in the recurrence table for a timeslot (timeRecurrences[id] for example). The reason for using both is to create a unique id for each component inside a recurrence. By combining the pk of the timeslot ("pk") with the id of the recurrence ("id") we can create an unique id, which can be assigned to the components that are being rendered inside a recurrence.

Will renaming "pk" to "timeslotPk" make it easier to understand what is going on? Can also add a comment about it in the code.

Reply:
I think that might be wise, especially as there's an overhaul needed in the database schema in the backend which might change which way the dependencies go. The frontend being more self-documenting would be a smart move.